src/fmeval/eval_algorithms/util.py

```diff
@@ -427,40 +415,58 @@ def create_model_invocation_pipeline(model: ModelRunner, prompt_template: str) -
 
-
-    :param pipeline: The evaluation algorithm's pipeline, to be executed on one or more datasets.
-    :param dataset: The dataset to execute the algorithm's pipeline on.
-    :param dataset_name: The name of the dataset.
+    :param dataset: The dataset to be evaluated.
+    :param pipeline: The evaluation algorithm's pipeline, to be executed on the dataset.
+    :param dataset_name: The name of the dataset being evaluated. This is metadata that
+        will be included in the returned EvalOutput object.
     :param eval_name: The name of the evaluation algorithm.
     :param metric_names: The names of the metrics that this evaluation algorithm computes.
-    :param eval_results_path: A file containing evaluation results will be stored under this directory.
-    :param prompt_template: The prompt template used to generate the prompts that are fed
-        to the model when it is invoked. Note that this argument is passed purely to be included
-        as metadata in the returned EvalOutput. The `pipeline` argument should already contain
-        the relevant prompt-generation and model-invocation transforms whose behavior is
-        configured by this argument. See the `evaluate_dataset` function for more details.
+        prior to performing any evaluation logic. This parameter is algorithm-specific.
+    :param eval_results_path: A file containing evaluation results will be stored at this path.
+    :param model: An instance of ModelRunner representing the model under evaluation.
+        If this argument is None, model responses cannot be obtained. In such cases,
+        the dataset configured by `dataset_config` should already contain a column for
+        model outputs.
+    :param prompt_template: A template used to generate prompts that are fed to the model.
+        If set to None, a default value will be used. Note that if this argument is not None,
+        `model` must also not be None.
     :param agg_method: The aggregation method to use when aggregating the computed metric values.
         Currently, only MEAN is supported.
     :param save: If set to true, prompt responses and scores will be saved to a file.
         The path that this file is stored at is configured by `eval_results_path`.
 
-    :return: A list of EvalOutput objects.
+    :return: An EvalOutput object encapsulating the results of the evaluation.
     """
+    if prompt_template:
+        util.require(model, "A model must be provided as well if the provided prompt template is not None.")
+    if model:
+        prompt_template = get_default_prompt_template(dataset_name) if not prompt_template else prompt_template
+        model_invocation_pipeline = create_model_invocation_pipeline(model, prompt_template)
+        pipeline = TransformPipeline([model_invocation_pipeline, pipeline])
+    else:
+        try:
+            validate_dataset(dataset, [DatasetColumns.MODEL_OUTPUT.value.name])
+        except EvalAlgorithmClientError:
```

Comment:
The try-except is just for providing a more specific error message than the generic one provided by `validate_dataset`